celery/concurrency/eventlet.py

```diff
@@ -144,3 +158,26 @@ def _get_info(self):
             'running-threads': self._pool.running(),
         })
         return info
+
+    @staticmethod
+    def _make_killable_target(target):
+        def killable_target(*args, **kwargs):
+            try:
+                return target(*args, **kwargs)
+            except GreenletExit:
+                return (False, None, None)
+            except Exception:
+                raise
+        return killable_target
+
+    def _add_to_pool_map(self, pid, green_thread):
+        self._pool_map[pid] = green_thread
+        green_thread.link(
+            TaskPool._cleanup_after_job_finish,
+            self._pool_map,
+            pid
+        )
+
+    @staticmethod
+    def _cleanup_after_job_finish(gt, pool_map, pid):
+        pool_map.pop(pid)
```

Comment:
```suggestion
        del pool_map[pid]
```

`dict.pop()` returns a value we don't need. We should use `del` instead as it will just remove the key.